minimesos/src/main/java/com/containersol/minimesos/docker/DockerClientFactory.java

```diff
@@ -3,28 +3,84 @@
 import com.github.dockerjava.api.DockerClient;
 import com.github.dockerjava.core.DockerClientBuilder;
 import com.github.dockerjava.core.DockerClientConfig;
-import org.apache.commons.lang.StringUtils;
+import java.io.File;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.github.dockerjava.core.DockerClientConfig.*;
 
 /**
  * Factory for creating {@link DockerClient}s
  */
 public class DockerClientFactory {
-
     private static DockerClient dockerClient;
+    private static DockerClientConfigBuilder configBuilder = new DockerClientConfigBuilder();
+    private static String dockerHostEnv;
+    private static String dockerCertPathEnv;
+    private static String dockerHostPrp;
+    private static String dockerCertPathPrp;
+    private static String dockerUri;
+    private static String dockerCertPath;
+    private static Properties properties = System.getProperties();
+    private static Map<String,String> environment = System.getenv();
 
     public static DockerClient build() {
-        if (dockerClient == null) {
-            DockerClientConfig.DockerClientConfigBuilder builder = DockerClientConfig.createDefaultConfigBuilder();
+        DockerClientConfig config = populateConfigBuilder(configBuilder).build();
+        dockerClient = DockerClientBuilder.getInstance(config).build();
+        return dockerClient;
+    }
 
-            String dockerHostEnv = System.getenv("DOCKER_HOST");
-            if (StringUtils.isBlank(dockerHostEnv)) {
-                builder.withUri("unix:///var/run/docker.sock");
-            }
+    public static DockerClient build(Properties argProperties) {
```

Comment:
Seems, it's either system properties or environment variables, but not a mix of them. Right?